### ISSUE TITLE
Executive Summaries: Attempts @ CoT monitor training

### ISSUE THREAD

**Issue by GnarlyMshtep** [GnarlyMshtep/pass_at_k#64]
I want to think of an actionable RL setup for monitorability training. 

* I need training data where the model is sometimes very monitorable and other times not, and ideally its a smooth gradient. Interpolating between the GDM hint setup seems like a good idea. 
* If we did interpolated-GDM-setup (IGS), what hint distribution would we use? 
  * If we use just correct hints, the model will learn to stop thinking, solve the hint / acknowledge it when it does not need solving. It already acknowledges the hint part of the time when it is simple, so this is very plausible. This is not very interesting. 
  *

**Comment by GnarlyMshtep** [comment]
## Somewhat incoherent about interpolated GDM setup 

### What is our goal? 
To get the LM to understand what factors made it reach its final decision and explicitly state them in its reasoning. More mathematically (maybe what we should focus on). We want to maximize our definition.  

### The IGS setup 
**be more precise about reward / how we determine ground truth of whether the hint was used**

**what should we have the monitor look out for? Previously we had it just look out for thinking about the hint, but now it seems we want to know whether the hint impacted the final answer, which is different.**

### Hint dist: all correct
Just do hint with no other reasoning. Also it wouldn't be that tricky to optimize, since the model already mentions the hint enough even in the simple setup. I think this does not require the model to do anything deep enough to actually chisel in new circuits

### Hint distr: all incorrect 
Ignore hint

### Hint distr: questions are hard enough and the hint is correct 
Must pay attention to hint, but judge whether you should use it, and state whether you did/

**Comment by GnarlyMshtep** [comment]
## Another idea: tapping into why the model uses a scratchpad 
Do a complex reward calculation which involves race so that it is declared in the scratchpad.

**Comment by GnarlyMshtep** [comment]
# Executive summary up 'till now [Oct 22, 2025]
#51 #56 are the key issues. 


### 1. Devising the experiment 
We devised the basic CoT monitorability setup: 
  1. within each batch, each question is asked as "baseline" (no hint) and hint which is either correct, incorrect, or omission (try to make the question somewhat subtly incomplete and give correct hint). 
  2. Advantage is computed by giving correctness + monitor calibration reward. 
  3. The monitor score is calibrated to 0 for for baseline and in the hinted case calibrated to the 'empirical true effect size' which is the difference in the hint selection rate between hinted and non-hinted answers.   

### 2.  Experiments 
**main tweaks**: `bigmath-digits` / `deepmath`, string matching / model monitor, correct hints only / all hint types, 'emperical effect size' / `did_sel_hint`,  $\lambda = 1$ /    $\lambda = .3$ weight on calibration objective weight. 

@andrewilyas Unfortunately, I don't exactly recall exactly what I ran w/o digging through some wandb configs. I try to give the main vibes but don't hesitate to ask me about missing info that is useful! I will also update with useful missing info as I think and explore it.  
   
1. Initially: optimization issues
1. optimization issues solved, but bigmath was easy and the Q2.5-7B was essentially starting at 70%. This means the effect size was small since it was mostly right on its own. The behavior we were seeing is that it said "hint" less. 
1. Switched to deepmath, which a student told me was harder. It was harder but the results were not different: we were seeing the optimized score go up (with calibration usually staying about the same for hinted questions) -- supposedly because the effect size was changing. 
1. iirc, same behavior on `deepmath` when using a string matching.
1. somewhere in here fixed `deepmath` reward, and **changed the 'true emperical effect size' of the hinted answer to 'did_sel_hint' -- literally whether it selected the hinted answer, it would be calibrated for saying "hint"**. Also added the $\lambda = 0.3$ weight. The next experiment was will all of those. 
1. deepmath, only correct hints, string matching, tried and tested optimization which is stable on deepmath with correctness reward only train for ~200 steps. We see similar results as before (hinted monitor score plummets immidiately) when up until step 100 when it spikes up. Our optimization is generally very unstable-looking. It's [these](https://wandb.ai/matan-shtepel-carnegie-mellon-university/cot-monitor/runs/h538mb4w?nw=nwusermatanshtepel) [two](https://wandb.ai/matan-shtepel-carnegie-mellon-university/cot-monitor/runs/4jpjlycx?nw=nwusermatanshtepel) runs. 
  1. iirc, I did not train on just `deepmath` on 7B model with our `deltaAI` config. It could be that the issue is not due to the reward. 
  1. idrk why the optimization decides to plummet the monitor
  1. one guess for part of the instability is that around step 40 the baseline reaches 93% correctness, and I noticed that these RL algos behave eraticly and sometimes just crash when you reach ~100% of the satisfiable reward. 

### 3. @GnarlyMshtep TLDR on current experiments
1. Working with emperical effect size seems unstable because the model can try to get more calibrated by shrinking it instead of saying "hint" more. Especially if the dataset is east and seemingly every dataset is easy :sad
1. For some reason it is a pretty consistent trend iirc across settings that initially (steps1-5) spikes, baseline hinted rate follows (steps5-10), then hinted monitor rate drasticly plummets and baseline monitor rate slowly declines. Also with $\lambda = 1$. 
1. I feel pretty optimistic on making all hint, string matching (or even model), `did_sel_hint` effect size, `deepmath` training work and relatively smoothly, although unsure about solving the issue in the bullet above this one.
